### DISCUSSION TITLE
domain events

### DISCUSSION THREAD

**Discussion by gpicavet** [techexcellenceio/techexcellence#18] [Ideas]
Hi Valentina,

Great job !

In complex applications, we often use domain events as it can help decrease coupling and better stick to Single Responsability Principle.

May i suggest to give a simple example in this kata ?

for exemple one technique i use is to store events in entity and using some transactional / after commit event publisher. 
But probably not great in all cases :)

**Comment by valentinajemuovic** [comment]
Hi @gpicavet, thank you! 

Yes, that's a great point, including domain events, and then for example for commands asserting that certain domain events were published.

> for exemple one technique i use is to store events in entity and using some transactional / after commit event publisher.

I like that technique too.

Thanks for your suggestion!

I'll write a comment here after I add this to the kata.
